explorer/testdata/operators/shortcircuit_and.carbon

```diff
@@ -0,0 +1,23 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// RUN: %{explorer} %s 2>&1 | \
+// RUN:   %{FileCheck} --match-full-lines --allow-unused-prefixes=false %s
+// RUN: %{explorer} --parser_debug --trace_file=- %s 2>&1 | \
+// RUN:   %{FileCheck} --match-full-lines --allow-unused-prefixes %s
+// AUTOUPDATE: %{explorer} %s
+// CHECK: result: 1
```

Comment:
Is there a good way to avoid defining 4 separate operators test data for the 4 use cases? Regrouping, or parametrizing would help, but didn't find any equivalent in `testdata`.

Reply:
Having four separate tests seems fine for now. We're generally leaning towards having lots of small tests instead of a smaller number of larger tests.